plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx

```diff
@@ -25,48 +25,39 @@ type StatusBadge = {
 	explanation?: string;
 };
 
-function getStatusBadges( subscription: Subscription ): StatusBadge[] {
-	const badges: StatusBadge[] = [];
-
-	if ( subscription.local.installed && ! subscription.active ) {
-		badges.push( {
-			text: __( 'Not connected', 'woocommerce' ),
-			level: StatusLevel.Warning,
-			explanation: __(
-				'To receive updates and support, please connect your subscription to this store.',
-				'woocommerce'
-			),
-		} );
-	}
-
+function getStatusBadge( subscription: Subscription ): StatusBadge | false {
 	if ( subscription.product_key === '' ) {
-		badges.push( {
+		/**
+		 * If there is no subscription, we don't need to check for the expiry.
+		 */
+		return {
 			text: __( 'No subscription', 'woocommerce' ),
 			level: StatusLevel.Error,
 			explanation: __(
 				'To get updates and support for this extension, you need to purchase a new subscription, or else share or transfer a subscription for this extension from another account.',
 				'woocommerce'
 			),
-		} );
-
-		/**
-		 * If there is no subscription, we don't need to check for the expiry.
-		 */
-		return badges;
-	}
-
-	if ( subscription.expired ) {
-		badges.push( {
+		};
+	} else if ( subscription.local.installed && ! subscription.active ) {
+		return {
```

Comment:
Considering we're returning immediately, we don't need the `else`. But maybe there was a bug I didn't realize when writing this code.

Reply:
Thanks!